tests/api_tests/nodes/test_views.py

```diff
@@ -863,6 +865,512 @@ def test_filtering_on_invalid_field(self):
+        self.project.set_visible(self.user_two, False, save=True)
+        self.project.reload()
+        
+        data = {
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_creator, data, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+        self.project.reload()
+        assert_true(self.project.get_visible(self.user))
+
+    def test_change_contributor_non_admin_auth(self):
+        data = {
+            'permission': permissions.READ,
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), [permissions.READ, permissions.WRITE])
+        assert_true(self.project.get_visible(self.user_two))
+
+    def test_change_contributor_not_logged_in(self):
+        data = {
+            'permission': permissions.READ,
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), [permissions.READ, permissions.WRITE])
+        assert_true(self.project.get_visible(self.user_two))
+
+
+class TestNodeContributorDelete(ApiTestCase):
+    def setUp(self):
+        super(TestNodeContributorDelete, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+        self.user_three = AuthUserFactory()
+
+        self.project = ProjectFactory(creator=self.user)
+        self.project.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+        self.project.reload()
+
+        self.url_user = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user._id)
+        self.url_user_two = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user_two._id)
+        self.url_user_three = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user_three._id)
+
+    def test_remove_contributor_admin(self):
+        res = self.app.delete(self.url_user_two, auth=self.user.auth)
+        assert_equal(res.status_code, 204)
+
+        self.project.reload()
+        assert_not_in(self.user_two, self.project.contributors)
+
+    def test_remove_contributor_non_admin_is_forbidden(self):
+        self.project.add_contributor(self.user_three, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+
+        res = self.app.delete(self.url_user_three, auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_in(self.user_three, self.project.contributors)
+
+    def test_remove_self_non_admin(self):
```

Comment:
Thanks for adding this test. Can we also get an assertion checking the status code from the request? (204, I think)